src/app/App.tsx

```diff
@@ -34,11 +34,38 @@ const Main = () => {
 
 	const isOnline = useNetworkStatus();
 
+	const syncDelegates = (env: Environment) => {
+		console.log("Running delegates sync...");
+		const coinsData = env.usedCoinsWithNetworks();
+		const coinsInUse = Object.keys(coinsData);
+		const delegatesPromises: any = [];
+
+		coinsInUse.forEach((coin) => {
+			const coinNetworks = coinsData[coin];
+			coinNetworks.forEach((network) => {
+				delegatesPromises.push(Promise.resolve(env.coins().syncDelegates(coin, network)));
+			});
+		});
+
+		Promise.allSettled(delegatesPromises).then((results: any) => {
+			results.forEach(({ status }: { status: string; value: any }) => {
+				if (status !== "fulfilled") throw new Error("Error synchronizing delegates");
```

Comment:
We shouldn't throw an error and abort evaluating the results just because an error occurred. I don't even see why we need to evaluate them at all. The only reason to use `allSettled` instead of `all` is so that we don't encounter an exception if 1 out of 20 coins fails to sync their data.